### ISSUE TITLE
No longer able to connect Last.fm session to the extension

### ISSUE THREAD

**Issue by bqback** [web-scrobbler/web-scrobbler#4351]
**Describe the bug**
The buttons to log into Last.fm/Libre.fm/ListenBrainz don't work

**How to reproduce**
1. Go to extension settings while not logged into any of the services
2. Open dev console in the browser
3. Click on the button to log into any of the services
4. See an uncaught error in promise

**Expected behavior**
A tab opens offering to log into a service.

**Screenshots**
![изображение](https://github.com/web-scrobbler/web-scrobbler/assets/28458777/77aa3a3e-d1a0-43df-ba2a-f4d9f4bb20bd)

**Debug logs**
Last.fm: Error while fetching session [context-menu-f70282e7.js:15:49694](moz-extension://372a2354-5b59-4d88-8d45-c9a9f203bf02/assets/context-menu-f70282e7.js)
Error: An unexpected error occurred [context-menu-f70282e7.js:15:49694](moz-extension://372a2354-5b59-4d88-8d45-c9a9f203bf02/assets/context-menu-f70282e7.js)
Libre.fm: Error while fetching session [context-menu-f70282e7.js:15:49694](moz-extension://372a2354-5b59-4d88-8d45-c9a9f203bf02/assets/context-menu-f70282e7.js)
Error: An unexpected error occurred [context-menu-f70282e7.js:15:49694](moz-extension://372a2354-5b59-4d88-8d45-c9a9f203bf02/assets/context-menu-f70282e7.js)
ListenBrainz: Error while fetching session [context-menu-f70282e7.js:15:49694](moz-extension://372a2354-5b59-4d88-8d45-c9a9f203bf02/assets/context-menu-f70282e7.js)
Error: An unexpected error occurred [context-menu-f70282e7.js:15:49694](moz-extension://372a2354-5b59-4d88-8d45-c9a9f203bf02/assets/context-menu-f70282e7.js)
Uncaught (in promise) Error: An unexpected error occurred undefined

**Environment (please complete the following information)**
-   OS: Windows
-   Browser: Firefox
-   Extension version: 3.2.0

**Comment by yayuyokitano** [comment]
Unable to reproduce. I know roughly where the error has to be occurring, but I'm not entirely sure where exactly.
Out of curiosity, what do you get if you run the foillowing in web scrobbler options page:
```ts
(await browser.storage.local.get("LastFM")).LastFM
```

**Comment by bqback** [comment]
> what do you get if you run the foillowing in web scrobbler options page

Nothing at all, it would seem

![изображение](https://github.com/web-scrobbler/web-scrobbler/assets/28458777/df0f94bf-3589-4c4b-8107-f7103298f3a2)

**Comment by bqback** [comment]
In the meantime I've tried disabling ALL extensions beside WS, no dice.

**Comment by yayuyokitano** [comment]
On further research I believe there is some sort of CORS-related issue affecting some firefox users, but it does not seem to affect everyone. Bit hard for me to test this as I am unable to reproduce, but looking into it.

**Comment by Mirika1** [comment]
I was experiencing the same issue with ListenBrainz on Microsoft Edge, and I found out that it was because I had the "This can read and change site data" option for the Web Scrobbler extension set to "When you click the extension". When I changed it to run on ListenBrainz, I was able to connect my account.

**Comment by VulcanSphere** [comment]
Can confirm with ListenBrainz, on Firefox.

I already signed in to ListenBrainz, and the authentication still throw error.

**Comment by squigglezworth** [comment]
This is still ongoing? 

Any fix yet?

**Comment by rob9315** [comment]
still ongoing, hard to reproduce

**Comment by gdressel** [comment]
As @Mirika1 suggested in https://github.com/web-scrobbler/web-scrobbler/issues/4351#issuecomment-1935277183: the permission on ListenBrainz website itself was blocking the integration for me as well.

<img width="613" height="498" alt="Image" src="https://github.com/user-attachments/assets/d468d6da-1a6f-420e-b5ca-53dd34dbb8d7" />

**Comment by CollapsingEden** [comment]
I think it has nothing to do with local storage actually.
I had the same exact problem and managed to log in after enabling a system-wide VPN instead of the usual split-tunneling i use.
So after looking into it some more i figured out that `ws.audioscrobbler.com` returns HTTP 403 for some reason. That seems to be the root cause of this problem.
I added this domain to my "this goes through VPN" list and now everything seems to be working fine.

**Comment by bqback** [comment]
> I think it has nothing to do with local storage actually. I had the same exact problem and managed to log in after enabling a system-wide VPN instead of the usual split-tunneling i use. So after looking into it some more i figured out that `ws.audioscrobbler.com` returns HTTP 403 for some reason. That seems to be the root cause of this problem. I added this domain to my "this goes through VPN" list and now everything seems to be working fine.

Would you happen to be residing in Russia or Kazakhstan?

**Comment by CollapsingEden** [comment]
> Would you happen to be residing in Russia or Kazakhstan?

Yes, i know about the geoblock thing. But at the same time i were able to scrobble tracks from my foobar2000 just fine without changing anything. So thats kinda confusing, it's the same API after all(?).

**Comment by rob9315** [comment]
there are a few different issues at play, if your login button did ANYTHING at all, then it wasn't this exact issue @CollapsingEden, probably the geoblock from #6030, yes

**Comment by CollapsingEden** [comment]
> there are a few different issues at play, if your login button did ANYTHING at all, then it wasn't this exact issue [@CollapsingEden](https://github.com/CollapsingEden), probably the geoblock from [#6030](https://github.com/web-scrobbler/web-scrobbler/issues/6030), yes

No, initially pressing the "Sing in" button seemingly did nothing at all, exactly as described here https://github.com/web-scrobbler/web-scrobbler/issues/6033 in the "other symtoms" section in relation to this issue. I also got the "Service error" thing.
But i guess it's not really the same EXACT problem, cause this issue is quite old and unlikely to be related to scrobller API access problems, which are relatively recent.
Sorry for the inconvenience.
